pkg/container-disk/container-disk_test.go

```diff
@@ -92,78 +91,98 @@ var _ = Describe("ContainerDisk", func() {
 				Entry("raw disk", "raw"),
 			)
 			It("by verifying error when no disk is present", func() {
-
-				vmi := api.NewMinimalVMI("fake-vmi")
-				appendContainerDisk(vmi, "r0")
+				vmi := libvmi.New(
+					libvmi.WithContainerDisk("r0", "someimage:v1.2.3.4"),
+				)
+				By("Ensuring no disk is present")
+				// Don't create any disk file to simulate the condition
+
+				By("Trying to get the volume mount directory on guest and expecting an error")
+				volumeMountDir := GetVolumeMountDirOnGuest(vmi)
+				_, err := os.Stat(filepath.Join(volumeMountDir, "disk_0.img"))
+				Expect(err).To(HaveOccurred())
+				Expect(errors.Is(err, os.ErrNotExist)).To(BeTrue())
 			})
 
 			It("by verifying host directory locations", func() {
-				vmi := api.NewMinimalVMI("fake-vmi")
+				vmi := libvmi.New(
+					libvmi.WithContainerDisk("r0", "someimage:v1.2.3.4"),
+				)
 				vmi.UID = "6789"
 				vmi.Status.ActivePods = map[types.UID]string{
 					"1234": "myhost",
 				}
 
 				// should not be found if dir doesn't exist
+				By("Checking if the directory does not exist")
 				path, err := GetVolumeMountDirOnHost(vmi)
 				Expect(err).To(HaveOccurred())
 				Expect(errors.Is(err, os.ErrNotExist)).To(BeTrue())
 
 				// should be found if dir does exist
+				By("Creating the expected path and verifying it exists")
 				expectedPath := fmt.Sprintf("%s/1234/volumes/kubernetes.io~empty-dir/container-disks", tmpDir)
 				Expect(os.MkdirAll(expectedPath, 0755)).To(Succeed())
 				path, err = GetVolumeMountDirOnHost(vmi)
 				Expect(err).ToNot(HaveOccurred())
 				Expect(unsafepath.UnsafeAbsolute(path.Raw())).To(Equal(expectedPath))
 
 				// should be able to generate legacy socket path dir
+				By("Generating and verifying the legacy socket path")
 				legacySocket := GetLegacyVolumeMountDirOnHost(vmi)
 				Expect(legacySocket).To(Equal(filepath.Join(tmpDir, "6789")))
 
 				// should return error if disk target dir doesn't exist
+				By("Verifying disk target directory")
 				targetPath, err := GetDiskTargetDirFromHostView(vmi)
 				expectedPath = fmt.Sprintf("%s/1234/volumes/kubernetes.io~empty-dir/container-disks", tmpDir)
 				Expect(err).ToNot(HaveOccurred())
 				Expect(unsafepath.UnsafeAbsolute(targetPath.Raw())).To(Equal(expectedPath))
-
 			})
 
 			It("by verifying error occurs if multiple host directory locations exist somehow", func() {
-				vmi := api.NewMinimalVMI("fake-vmi")
+				vmi := libvmi.New(
```

Comment:
The original VMI did not have a containerdisk.

Reply:
yes you are right i just replaced it with 
```go
vmi := libvmi.New()
```
and still working
i got confused with name i thought i will get null error if i did not define the  containerdisk